lib/auth/webauthn/messages_test.go

```diff
@@ -56,3 +57,270 @@ func TestCredentialAssertionResponse_json(t *testing.T) {
 	}
 }
+
+func TestCredentialCreation_Validate(t *testing.T) {
+	okCC := &wanlib.CredentialCreation{
+		Response: protocol.PublicKeyCredentialCreationOptions{
+			Challenge: make([]byte, 32),
+			RelyingParty: protocol.RelyingPartyEntity{
+				ID: "example.com",
+				CredentialEntity: protocol.CredentialEntity{
+					Name: "Teleport",
+				},
+			},
+			Parameters: []protocol.CredentialParameter{
+				{Type: protocol.PublicKeyCredentialType, Algorithm: webauthncose.AlgES256},
+			},
+			AuthenticatorSelection: protocol.AuthenticatorSelection{
+				UserVerification: protocol.VerificationDiscouraged,
+			},
+			Attestation: protocol.PreferNoAttestation,
+			User: protocol.UserEntity{
+				CredentialEntity: protocol.CredentialEntity{
+					Name: "llama",
+				},
+				DisplayName: "Llama",
+				ID:          []byte{1, 2, 3, 4, 5}, // arbitrary
+			},
+		},
+	}
+
+	tests := []struct {
+		name           string
+		createCC       func() *wanlib.CredentialCreation
+		alwaysCreateRK bool
+		wantErr        string
+	}{
+		{
+			name:     "ok", // check that good params are good
+			createCC: func() *wanlib.CredentialCreation { return okCC },
+			wantErr:  "",
+		},
+		{
+			name:     "nil cc",
+			createCC: func() *wanlib.CredentialCreation { return nil },
+			wantErr:  "credential creation required",
+		},
+		{
+			name: "cc without challenge",
+			createCC: func() *wanlib.CredentialCreation {
+				cp := *okCC
+				cp.Response.Challenge = nil
+				return &cp
+			},
+			wantErr: "challenge",
+		},
+		{
+			name: "cc without RPID",
+			createCC: func() *wanlib.CredentialCreation {
+				cp := *okCC
+				cp.Response.RelyingParty.ID = ""
+				return &cp
+			},
+			wantErr: "relying party ID",
+		},
+		{
+			name: "empty RP name",
+			createCC: func() *wanlib.CredentialCreation {
+				cp := *okCC
+				cp.Response.RelyingParty.Name = ""
+				return &cp
+			},
+			wantErr: "relying party name",
+		},
+		{
+			name: "rrk empty user name",
+			createCC: func() *wanlib.CredentialCreation {
+				cp := *okCC
+				cp.Response.User.Name = ""
+				return &cp
+			},
+			wantErr: "user name",
+		},
+		{
+			name: "rrk empty user display name",
+			createCC: func() *wanlib.CredentialCreation {
+				cp := *okCC
+				cp.Response.User.DisplayName = ""
+				return &cp
+			},
+			wantErr: "user display name",
+		},
+		{
+			name: "user ID",
+			createCC: func() *wanlib.CredentialCreation {
+				cp := *okCC
+				cp.Response.User.ID = nil
+				return &cp
+			},
+			wantErr: "user ID",
+		},
+		{
+			name: "cc without rp name",
+			createCC: func() *wanlib.CredentialCreation {
+				cp := *okCC
+				cp.Response.RelyingParty.Name = ""
+				return &cp
+			},
+			wantErr: "relying party name required",
+		},
+	}
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			err := test.createCC().Validate()
+			if test.wantErr != "" {
+				require.ErrorContains(t, err, test.wantErr, "Validate returned err = %q, want %q", err, test.wantErr)
```

Comment:
Keeping the same Errorf strings in a `require` check doesn't make sense, this is going to print a bunch of duplicate information now.

```suggestion
				require.ErrorContains(t, err, test.wantErr, "Validate returned unexpected error")
```